src/store/messages/saga.ts

```diff
@@ -615,31 +624,41 @@ const inProgress = {};
 export function* loadAttachmentDetails(action) {
   const { media, messageId } = action.payload;
 
-  if (inProgress[messageId] || media.url) {
+  // Check conditions to skip processing
+  if (inProgress[messageId] || media.url || media.downloadStatus === MediaDownloadStatus.Failed) {
```

Comment:
In follow up PR, we won't even call this `loadAttachmentDetails` if status === failed, but adding check here to double up on handling the errors